src/Microsoft.AspNet.Razor.Runtime/BufferEntryCollection.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Represents a hierarchy of strings and provides an enumerator that iterates over it as a sequence.
```

Comment:
nit: avoid word "represents" everywhere.  this is just `/// An ordered collection of strings.`